code/lib/test/src/spy.ts

```diff
@@ -1,17 +1,60 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import type { MockInstance } from '@vitest/spy';
 import {
-  spyOn,
+  spyOn as vitestSpyOn,
   isMockFunction,
-  fn,
+  fn as vitestFn,
   mocks,
   type MaybeMocked,
   type MaybeMockedDeep,
   type MaybePartiallyMocked,
   type MaybePartiallyMockedDeep,
 } from '@vitest/spy';
+import type { SpyInternalImpl } from 'tinyspy';
+import * as tinyspy from 'tinyspy';
 
 export type * from '@vitest/spy';
 
-export { spyOn, isMockFunction, fn, mocks };
+export { isMockFunction, mocks };
+
+type Listener = (mock: MockInstance, args: unknown[]) => void;
+let listeners: Listener[] = [];
+
+export function onMockCall(callback: Listener): () => void {
+  listeners = [...listeners, callback];
+  return () => {
+    listeners = listeners.filter((listener) => listener !== callback);
+  };
+}
```

Comment:
nit: to me it seems more like `listeners` is a Set.

```suggestion
type Listener = (mock: MockInstance, args: unknown[]) => void;
const listeners = new Set<Listener>();

export function onMockCall(callback: Listener): () => void {
  listeners.add(callback);
  return () => {
    listeners.delete(callback);
  }
}
```

Reply:
I like that